### DISCUSSION TITLE
HMR issue with Storybook + Vite + MJML

### DISCUSSION THREAD

**Discussion by telam** [storybookjs/storybook#26984] [Help]
### Summary

Hello,

I'm using the following libraries to test emails in Storybook:
- "@faire/mjml-react": "^3.3.0",
- "@rollup/pluginutils": "^5.1.0",
- "mjml": "^4.15.3",
- "mjml-browser": "^4.15.3",
- "react": "^18.3.1",
- "react-dom": "^18.3.1",
- "storybook": "^8.0.9"
- "vite": "^5.2.10"

I have the emails showing up properly in Storybook with the use of a custom decorator that looks similar to this:

```typescript jsx
const mjml = renderToMjml(storyFn(storyContext));
const { html, error } = mjml2html(mjml, {
  validationLevel: "soft",
});
return <div dangerouslySetInnerHTML={{ __html: html }} />;
```

However, anytime I make a change to my React component Storybook will not re-render the component. I can see the logs from Vite that HMR is working and if I click on the new .tsx file in the developer tools I can see the new file recognized by the browser with my change.

### Additional information

**main.ts**
```typescript
import type { StorybookConfig } from "@storybook/react-vite";

const config: StorybookConfig = {
  stories: ["../src/**/*.mdx", "../src/**/*.stories.@(js|jsx|mjs|ts|tsx)"],
  addons: [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    "@chromatic-com/storybook",
    "@storybook/addon-interactions",
  ],
  framework: {
    name: "@storybook/react-vite",
    options: {},
  },
  docs: {
    autodocs: "tag",
  },
};
export default config;

```

**preview.ts**
```typescript
import type { Preview } from "@storybook/react";
import { renderToMjml } from "@faire/mjml-react/utils/renderToMjml";
import mjml2html from "mjml-browser";

const preview: Preview = {
  parameters: {
    controls: {
      matchers: {
        color: /(background|color)$/i,
        date: /Date$/i,
      },
    },
  },
  decorators: [
    (Story) => {
      const mjml = renderToMjml(<Story />);
      const { html, error } = mjml2html(mjml, {
        validationLevel: "soft",
      });
      return <div dangerouslySetInnerHTML={{ __html: html }} />;
    },
  ],
};

export default preview;
```

### Create a reproduction

https://stackblitz.com/edit/github-tgehw4?file=src%2Fstories%2FExample.tsx

**Comment by valentinpalkovic** [comment]
Closing in favour of https://github.com/storybookjs/storybook/issues/26985
